pkg/controllers/webhookconfig/webhookconfig.go

```diff
@@ -126,6 +131,16 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options)
 }
 
 func (r *Reconciler) ReadyCheck(_ *http.Request) error {
+	// skip readiness check if we're not leader
+	// as we depend on caches and being able to reconcile Webhooks
+	if !r.ready {
+		select {
+		case <-r.leaderChan:
+			r.ready = true
+		default:
+			return nil
+		}
+	}
 	r.rdyMu.Lock()
```

Comment:
@moolen Ah I think you'll need to put these locks before the above ready check.

And then remove the below ready check, right? Since it's no longer needed?

Reply:
```

Ran 3 of 3 Specs in 30.983 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAPIs (30.98s)
PASS
```

Ran the test with race detection after the above mentioned fix. Looks okay.